Makefile

```diff
@@ -117,23 +116,55 @@ help-install:
 	@echo " Note that building a package in release mode ignores other packages present in"
 	@echo " the worktree. See Dune documentation for more information."
 
+# Generated Dune files for the test suite
+DUNE_TEST_GEN=\
+	_build/default/test_suite_dune\
+	_build/default/utest_dune\
+
+# Generated Dune files for the stdlib
+DUNE_GEN=\
+	_build/default/theories_dune\
+	_build/default/ltac2_dune\
+
 # We regenerate always as to correctly track deps, can do better
 # We do a single call to dune as to avoid races and locking
-_build/default/theories_dune _build/default/ltac2_dune .dune-stamp: FORCE
-	dune build $(DUNEOPT) --root . theories_dune ltac2_dune
-	touch .dune-stamp
+$(DUNE_GEN) .dune-stamp: FORCE
+	dune build $(DUNEOPT) --root . @dune-gen
+	@touch .dune-stamp
+
+# This target is called by testing targets. This will run the test suite rule
+# generator as awell as the stdlib one
+$(DUNE_TEST_GEN) .dune-test-stamp: FORCE
+	dune build $(DUNEOPT) --root . @dune-test-gen
+	@touch .dune-test-stamp
 
 theories/dune: .dune-stamp
-	cp -a _build/default/theories_dune $@ && chmod +w $@
+	@cp -a _build/default/theories_dune $@ && chmod +w $@
 
 user-contrib/Ltac2/dune: .dune-stamp
-	cp -a _build/default/ltac2_dune $@ && chmod +w $@
+	@cp -a _build/default/ltac2_dune $@ && chmod +w $@
+
+test-suite/dune: .dune-test-stamp
+	@cp -a _build/default/test_suite_dune $@ && chmod +w $@
+
+test-suite/unit-tests/dune: .dune-test-stamp
+	@cp -a _build/default/utest_dune $@ && chmod +w $@
 
 FORCE: ;
 
-DUNE_FILES=theories/dune user-contrib/Ltac2/dune
+DUNE_FILES=\
+	theories/dune\
+	user-contrib/Ltac2/dune\
+
+DUNE_TEST_FILES=\
+	test-suite/dune\
+	test-suite/unit-tests/dune\
 
 dunestrap: $(DUNE_FILES)
+	@touch test-suite/dune
+	@touch test-suite/unit-tests/dune
```

Comment:
Having empty dune files in the test suite seems like it will produce confusion.
Why not merge dunestrap and duneteststrap?

Reply:
I originally had both rules generated at the same time, but this will slow down the main build. I therefore opted to make them independent. We might be able to merge dunestrap and duneteststrap.

In this case, it might be possible to remove touching these files here, I will try it out.